client/src/store/utils.ts

```diff
@@ -153,3 +155,27 @@ export function parseSubtitlesToState(titles: IngestTitle): SubtitleFields {
 
     return result;
 }
+
+export function enableSceneGenerateCheck(metadata: ModelFields, unitsEntries: VocabularyOption[], purposeEntries: VocabularyOption[], typeEntries: VocabularyOption[]): boolean {
+    const { units, purpose, modelFileType } = metadata;
+    if (!units || !purpose || !modelFileType) return false;
+    const eVocabUnitSet = new Set([eVocabularyID.eModelUnitsMillimeter as number, eVocabularyID.eModelUnitsCentimeter, eVocabularyID.eModelUnitsMeter, eVocabularyID.eModelUnitsInch, eVocabularyID.eModelUnitsFoot, eVocabularyID.eModelUnitsYard]);
```

Comment:
Given that `eVocabUnitSet`, `eVocabPurposeSet`, and `eVocabFileTypeSet` are const and never change, perhaps these should be defined at the module level and not inside the function, to ensure they are evaluated just once, and not every time this method is called.

I'm not sure how javascript optimization works here -- perhaps function-scoped const items are evaluated just once. But I suspect they are evaluated each time the method executes.